adapters/gumgum/gumgum.go

```diff
@@ -127,25 +119,47 @@ func (g *GumGumAdapter) MakeBids(internalRequest *openrtb.BidRequest, externalRe
 	return bidResponse, errs
 }
 
-func preprocess(imp *openrtb.Imp) (string, error) {
+func preprocess(imp *openrtb.Imp) (*openrtb_ext.ExtImpGumGum, error) {
 	var bidderExt adapters.ExtImpBidder
 	if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
 		err = &errortypes.BadInput{
 			Message: err.Error(),
 		}
-		return "", err
+		return nil, err
 	}
 
 	var gumgumExt openrtb_ext.ExtImpGumGum
 	if err := json.Unmarshal(bidderExt.Bidder, &gumgumExt); err != nil {
 		err = &errortypes.BadInput{
 			Message: err.Error(),
 		}
-		return "", err
+		return nil, err
+	}
+
+	if imp.Banner != nil {
+		bannerCopy := *imp.Banner
+		if bannerCopy.W == nil && bannerCopy.H == nil && len(bannerCopy.Format) > 0 {
+			format := bannerCopy.Format[0]
+			bannerCopy.W = &(format.W)
+			bannerCopy.H = &(format.H)
+		}
+		imp.Banner = &bannerCopy
+	} else if imp.Video != nil {
+		err := validateVideoParams(imp.Video)
+		if err != nil {
+			return nil, err
+		}
+
+		if gumgumExt.Irisid != "" {
+			videoExt := openrtb_ext.ExtImpGumGumVideo{Irisid: gumgumExt.Irisid}
+			imp.Video.Ext, err = json.Marshal(&videoExt)
```

Comment:
You need to make a copy of `imp.Video` for setting a new `Ext`.

Reply:
Thanks!! I updated the code to make a copy before trying to set the new Ext. Please let me know if I need to modify it further